src/main/java/org/reaktivity/specification/socks/internal/SocksFunctions.java

```diff
@@ -126,6 +227,78 @@ public String getPrefixName()
         }
     }
 
+    private static void fillInBytes(
+        byte[] addressBytes,
+        int index,
+        String groupHex)
+    {
+        short groupBytes = parseShort(groupHex, 16);
+        addressBytes[index++] = (byte) ((groupBytes >> 8) & 0xff);
+        addressBytes[index] = (byte) (groupBytes & 0xff);
+    }
+
+    private static void encodeCompressedIpv6AddressBytes(
+        Matcher ipv6Matcher,
+        byte[] addressBytes)
+    {
+        Arrays.fill(addressBytes, (byte) 0);
+        for (int group = 1; group < 8; group++)
+        {
+            String groupHex = ipv6Matcher.group(group);
+            if (groupHex == null)
+            {
+                break;
+            }
+            fillInBytes(addressBytes, 2 * (group - 1), groupHex);
+        }
+        for (int group = 15; group > 8; group--)
+        {
+            String groupHex = ipv6Matcher.group(group);
+            if (groupHex == null)
+            {
+                break;
+            }
+            fillInBytes(addressBytes, 2 * (group - 8), groupHex);
+        }
+    }
+
+    public static byte parseByte(
```

Comment:
This method should not need to be `public`.
Please see http://tutorials.jenkov.com/java/access-modifiers.html and then select the appropriate access modifier for your needs.

Reply:
I updated with `protected` modifiers, because I want to test it which is outside the class but inside the package.